pkg/virtual/workspaces/registry/rest.go

```diff
@@ -549,3 +570,124 @@ func (s *REST) Delete(ctx context.Context, name string, deleteValidation rest.Va
 
 	return nil, false, errorToReturn
 }
+
+type KubeconfigSubresourceREST struct {
+	mainRest *REST
+
+	// coreClient is useful to get secrets
+	coreClient corev1client.CoreV1Interface
+	// workspaceShardClient can get KCP workspace shards
+	workspaceShardClient tenancyclient.WorkspaceShardInterface
+}
+
+var _ rest.Getter = &KubeconfigSubresourceREST{}
+var _ rest.Scoper = &KubeconfigSubresourceREST{}
+
+// Get retrieves a Workspace KubeConfig by workspace name
+func (s *KubeconfigSubresourceREST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	workspace, err := s.mainRest.Get(ctx, name, options)
+	if err != nil {
+		return nil, err
+	}
+	wrapError := func(err error) error {
+		k8sErr := kerrors.NewNotFound(tenancyv1alpha1.SchemeGroupVersion.WithResource("workspaces/kubeconfig").GroupResource(), name)
+		k8sErr.Status().Details.Causes = append(k8sErr.Status().Details.Causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeUnexpectedServerResponse,
+			Message: err.Error(),
+		})
+		return k8sErr
+	}
+	scope := ctx.Value(WorkspacesScopeKey).(string)
+	if workspace, isAWorkspace := workspace.(*tenancyv1alpha1.Workspace); isAWorkspace {
+		if conditions.IsTrue(workspace, tenancyv1alpha1.WorkspaceURLValid) {
+			ctx := genericapirequest.WithCluster(context.TODO(), genericapirequest.Cluster{Name: workspace.ClusterName})
```

Comment:
a new context is odd here. Can't we use `ctx`?

Reply:
Fixed in commit 2c97666d51d66ccc901f7d6f92992f5ba1412a5d